invenio_matcher/api.py

```diff
@@ -61,8 +64,12 @@ def match(record, queries=None, validator=None, **kwargs):
                                  ' MATCHER_QUERIES.')
 
     if not validator:
-        def validator(record, result):
-            return True
+        def validator(record, result, existing_matches={}):
```

Comment:
By that I mean that the "hack" does make the values retain within one `match` call, by the default argument gotcha. Perhaps not the most elegant way, so we could keep the state elsewhere.